### DISCUSSION TITLE
Upgrading Teslamate in Synology Docker - Questions

### DISCUSSION THREAD

**Discussion by ewrible** [teslamate-org/teslamate#2472] [Q&A] [answered]
So I've been using Teslamate for a while and semi-novice to Docker.  That said, I have running Teslamate on my Synology NAS.  When I first installed TM on a docker, this view was created.
![teslamate](https://user-images.githubusercontent.com/89223900/153906896-515f26c5-3dad-4b9d-8a7e-bcc82d53a5d5.jpg)

So when I get a notice to update Teslamate, I go ahead and download the new version in the Synology GUI, stop the container, duplicate it, etc. following these instructions:

https://www.wundertech.net/how-to-update-docker-containers-on-a-synology-nas/

But what's happening is that things like Grafana aren't getting updated via this update process.  I'm on 1.26.1 of Teslamate, but Grafana is still on version 7.5.10.  

Anyone running a Teslamate in a Synology docker and can help with the update process?  Thanks.

**Comment by ewrible** [comment]
Anyone have a thought?

**Comment by ewrible** [comment, marked as answer]
Alright, I think I got it.  Just me learning how docker works, etc.  This is resolved.

**Reply by fischeor** [reply]
So, how was the answer? I am facing the same problem and if you have a hint, it would be greatly appreciated. Thanks

**Reply by boba51** [reply]
Me too, after running successfully for a couple of years thanks to @jeffwrusso, I forgot how complicated it was to set up and so tried updating the docker images via the GUI, this, of course, has broken everything, I'd like to get it running and recover the data I've been storing.
I only really use Docker on Synology for this and so since the install hasn't touched it and would appreciate any suggestions fro recovering Teslamate. 
Happy to provide logs etc if anyone could point me in the right direction, it would be appreciated.

**Reply by jeffwrusso** [reply]
I also use Synology and when I upgrade Teslamate I use the CLI and follow these instructions: https://docs.teslamate.org/docs/upgrading/

**Reply by boba51** [reply]
@jeffwrusso Hi Yes, but being silly I just used the Synology GUI and now it is broken, I was hoping someone might be able to give me some tips on recovering it as I have several years of data I don't want to lose it I can help it.

**Reply by jeffwrusso** [reply]
I seem to recall I did something similar a bit ago and broke my install.  I then ran the upgrade commands via the cli and everything worked again.

**Reply by boba51** [reply]
@jeffwrusso Thanks, I think I've tried this already, but will give it another go :)

**Reply by boba51** [reply]
@jeffwrusso Thanks, seems to have improved things, but still behaving strangely, I had to login to Graphana using admin/admin which I believe are the default credentials and don't line up with those in the docker-compose.yml, I then have to manually enter the Database password from the docker-compose.yml and I can then see my cars past information, but it doesn't look like Teslamate is connected to the car so more work to be done.

**Reply by jeffwrusso** [reply]
The grafana thing is a change and to be expected now. I think at some point I updated my yml file as well to the latest one posted in the support docs.

**Reply by jeffwrusso** [reply]
Does your docker file contain the encryption key parameter?

https://docs.teslamate.org/docs/installation/docker

**Reply by boba51** [reply]
@jeffwrusso I've added my previous db password and encryption key parameter to a new docker-compose.yml file, re-run  "docker-compose up", there was/is a lot going on in the SSH window, but early signs are promising, I'm seeing live data from the car and my previous history, just waiting for the SSH to finish its stuff then will know for sure.
Thanks again for your help.

**Reply by boba51** [reply]
@jeffwrusso Looking good and hopefully, I've learned a valuable lesson, but we'll see ;)

![image](https://github.com/adriankumpf/teslamate/assets/17386935/e84478c0-9121-43c3-8fd6-96f8e89c4adb)

**Comment by dukefaenza** [comment]
i did all the steps and i get Cannot start service teslamate: Bind mount failed: '/volume1/@docker/import' does not exists
all containers start but docket_teslamate1
help

**Reply by boba51** [reply]
Sorry I can't be of much help as I've only managed to get it working by asking for help, but one of the things I did was copied a new docker-compose.yml file and copied across the unique things like passwords, dB location etc and rerun the "docker-compose up" and it seemed to do the job.
I did back up, update and restore the dB after messing things up originally by trying the update using the Synology GUI.

**Reply by jeffwrusso** [reply]
I'm not someplace I can test, but that error is because of a volume mapping that is missing from your docker.  You can either remove this from the yml file:`volumes:
      - ./import:/opt/app/import ` or create that folder at /volume1/@docker/import. That folder is only needed to import historical data.

**Comment by dukefaenza** [comment]
I don’t know why I didn’t t￼hink about that, I created the import folder and that works, now I have another issue docker_database_1 keep restarting after few seconds, 


Claudio Faenza
Saint Francis Catholic Academy | 
IT Director
4100 NW 115th Terrace
Gainesville FL 32606
